### PR TITLE
Add `Plugin_Review_PHPCS_Check` static check

### LINKED ISSUE #180: Implement plugin_review_PHPCS_check
### Is your enhancement related to a problem? Please describe.

https://github.com/WordPress/plugin-check/blob/trunk/checks/phpcs.php (we should call that more specifically something like Plugin_Review_PHPCS_Check, and it should use the existing Abstract_PHP_CodeSniffer_Check base class with the https://github.com/WordPress/plugin-check/blob/trunk/checks/phpcs/plugin-check.xml ruleset, which we can just copy to our repo)

### Acceptance Criteria
- The `Plugin_Review_PHPCS_Check` class should be created and located at `includes/Checker/Checks`.
- The `Plugin_Review_PHPCS_Check` class should extend the `Abstract_PHP_CodeSniffer_Check` abstract class.
- Create a new folder named `phpcs-rulesets` at the root level.
- Please copy the ruleset file `https://github.com/WordPress/plugin-check/blob/trunk/checks/phpcs/plugin-check.xml` into a new file named `plugin-review.xml`. Afterward, add the `plugin-review.xml` file to the `phpcs-rulesets` directory.
- Class should implement the public `get_args()` and return an associative array of the required PHPCS arguments
    - `'extensions' => 'php',`
    - `'standard'   => WP_PLUGIN_CHECK_PLUGIN_DIR_PATH . 'phpcs-rulesets/plugin-review.xml',`

### Test Coverage
- `run()` method adds a correct message to the `$check_result` when checking an invalid file
   - Use any deprecated functions from https://codex.wordpress.org/Category:Deprecated_Functions
   - Use PHP short tag. `<? echo 'Test'; ?>`
- `run()` method does not add any message to the `$check_result` when checking a valid file

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
Closes #180

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by mukeshpanchal27** [comment]
We are encountering a similar issue to what we faced in the previous pull request https://github.com/10up/plugin-check/pull/186#issuecomment-1589109005. The problem we are facing now is the same as the one mentioned in the linked comment.

File: includes/Checker/Checks/Plugin_Review_PHPCS_Check.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	 * Returns an associative array of arguments to pass to PHPCS.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array An associative array of PHPCS CLI arguments.
+	 */
+	protected function get_args() {
+		return array(
+			'extensions' => 'php',
+			'standard'   => WP_PLUGIN_CHECK_PLUGIN_DIR_PATH . 'phpcs-rulesets/plugin-review.xml',
```
Comment (spacedmonkey) [inline includes/Checker/Checks/Plugin_Review_PHPCS_Check.php:27]: Could be nice if this could be filtered or otherwise changed.

Comment (mukeshpanchal27) [reply includes/Checker/Checks/Plugin_Review_PHPCS_Check.php:27]: Thank you, @spacedmonkey, for your feedback.

This test is focused on meeting the requirements of the plugin review team, as outlined in https://github.com/WordPress/plugin-check/blob/trunk/checks/phpcs.php. Therefore, it is important for us to align with their specific requirements.

cc. @felixarntz

**Review by spacedmonkey — APPROVED** [review]

**Review by jjgrainger — COMMENTED** [review]
Thanks @mukeshpanchal27 looks good, approved

**Review by jjgrainger — APPROVED** [review]

File: tests/phpunit/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+use WordPress\Plugin_Check\Checker\Check_Context;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Checks\Plugin_Review_PHPCS_Check;
+use WordPress\Plugin_Check\Test_Utils\TestCase\Static_Check_UnitTestCase;
+
+class Plugin_Review_PHPCS_Check_Tests extends Static_Check_UnitTestCase {
+
+	public function test_run_with_errors() {
+		$plugin_review_phpcs_check = new Plugin_Review_PHPCS_Check();
+		$check_context             = new Check_Context( TESTS_PLUGIN_DIR . '/tests/phpunit/testdata/plugins/test-plugin-review-phpcs-errors/load.php' );
```
Comment (felixarntz) [inline tests/phpunit/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php]: Let's use the new constant introduced in #189 here and below.

File: tests/phpunit/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		// Check for WordPress.WP.DeprecatedFunctions.the_author_emailFound error on Line no 12 and column no at 5.
+		$this->assertArrayHasKey( 12, $errors['load.php'] );
+		$this->assertArrayHasKey( 5, $errors['load.php'][12] );
+		$this->assertArrayHasKey( 'code', $errors['load.php'][12][5][0] );
+		$this->assertEquals( 'WordPress.WP.DeprecatedFunctions.the_author_emailFound', $errors['load.php'][12][5][0]['code'] );
+	}
+
+	public function test_run_without_errors() {
+		$plugin_review_phpcs_check = new Plugin_Review_PHPCS_Check();
+		$check_context             = new Check_Context( TESTS_PLUGIN_DIR . '/tests/phpunit/testdata/plugins/test-plugin-review-phpcs-without-errors/load.php' );
```
Comment (felixarntz) [inline tests/phpunit/Checker/Checks/Plugin_Review_PHPCS_Check_Tests.php]: See above.

**Review by felixarntz — APPROVED** [review]
@mukeshpanchal27 Approving this already, though it needs to have `trunk` merged back to avoid the merge conflict. Related, please also update the below after you've merged. Great work!

File: phpcs-rulesets/plugin-review.xml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	<exclude-pattern>*/cmb2/*</exclude-pattern>
+	<exclude-pattern>*/redux-framework/*</exclude-pattern>
+	<exclude-pattern>*/cherry-framework/*</exclude-pattern>
+	<exclude-pattern>*/titan-framework/*</exclude-pattern>
+	<exclude-pattern>*/vendor/*</exclude-pattern>
+	<exclude-pattern>*/guzzlehttp/*</exclude-pattern>
+	<exclude-pattern>*/vendors/*</exclude-pattern>
+	<exclude-pattern>*/plugin-update-checker/*</exclude-pattern>
+	<exclude-pattern>*/composer_directory/*</exclude-pattern>
+	<exclude-pattern>*/node_modules/*</exclude-pattern>
```
Comment (sybrew) [inline phpcs-rulesets/plugin-review.xml:31]: Developers could hide backdoors in these files as they go untested.

**Pull request merged**